api/src/domain/tax-filings/taxFilingsFactory.ts

```diff
@@ -0,0 +1,43 @@
+import { UserData } from "@shared/userData";
+import { ApiTaxFilingClient, TaxFilingClient, UserDataClient } from "../types";
+
+export const taxFilingsFactory = (
+  userDataClient: UserDataClient,
+  taxFilingClient: ApiTaxFilingClient
+): TaxFilingClient => {
+  const lookup = async (props: {
+    userId: string;
+    taxId: string;
+    businessName: string;
+  }): Promise<UserData> => {
+    let userData = await userDataClient.get(props.userId);
+    const response = await taxFilingClient.lookup(props.taxId, props.businessName);
+    userData =
+      response.state == "SUCCESS"
+        ? { ...userData, taxFilingData: { ...userData.taxFilingData, ...response } }
+        : { ...userData, taxFilingData: { ...userData.taxFilingData, state: response.state } };
```

Comment:
don't these two cases from the ternary result in the same output? either way, it's using the state from the response?